src/lib/discovery/dns.ts

```diff
@@ -34,14 +41,22 @@ export class DnsNodeDiscovery {
    * return fewer peers than requested if `maxQuantity` is larger than the number
    * of ENR records or the number of errors/duplicate peers encountered by randomized
    * search exceeds `maxQuantity` plus the `errorTolerance` factor.
+   * Will return peers only with certain capabilities (protocols) enabled if
+   *  `wantedNodeCapabilityCount` object is passed in.
    */
-  async getPeers(maxQuantity: number, enrTreeUrls: string[]): Promise<ENR[]> {
+  async getPeers(
+    maxQuantity: number,
+    enrTreeUrls: string[],
+    wantedNodeCapabilityCount: NodeCapabilityCount | undefined
+  ): Promise<ENR[]> {
     let totalSearches = 0;
     const peers: ENR[] = [];
 
     const networkIndex = Math.floor(Math.random() * enrTreeUrls.length);
     const { publicKey, domain } = ENRTree.parseTree(enrTreeUrls[networkIndex]);
 
+    console.log("wantedNodeCapabilityCount:", wantedNodeCapabilityCount);
```

Comment:
I was stuck in here on how to match the node (peer?) to the `wantedNodeCapabilityCount`. I tried keeping a count but couldn't find an elegant way to keep track of the multiple counts.